File: LeaveManegamentProject/LeaveManegamentProject/Controllers/AdminController.cs
using LeaveManegamentProject.DbContexts;
using LeaveManegamentProject.Helper;
using LeaveManegamentProject.ViewModels;
using LeaveManegamentProject.Entities;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace LeaveManegamentProject.Controllers
{
    public class AdminController : Controller
    {
        LeaveContext context = new LeaveContext();
        Response response = new Response();

        [HttpGet]
        public string Index()
        {
            var adminlist = context.Admins.ToList();
            return JsonConvert.SerializeObject(AutoMapper.Mapper.Map<AdminView[]>(adminlist));
        }

        [HttpPost]
        public string Login(string Email, string Password)
        {
            var admin = context.Admins.Find(Email);

            if (string.IsNullOrEmpty(Email))
            {
                response.Message = "Please Provide Email Id";
                response.IsSuccess = false;
                return JsonConvert.SerializeObject(response);
            }
            if (string.IsNullOrEmpty(Password))
            {
                response.Message = "Please Provide Password Id";
                response.IsSuccess = false;
                return JsonConvert.SerializeObject(response);
            }
            if (context.Admins.Any(c => c.Email == Email) && admin.Password == Password)
            {
                response.Message = "Login Successfully";
                response.IsSuccess = true;
                response.Data = admin;
                return JsonConvert.SerializeObject(response);
            }

            response.Message = "check Your Email & Password";
            response.IsSuccess = false;
            return JsonConvert.SerializeObject(response);
        }
        [HttpPost]
        public string Save(AdminView adminview)
        {
            AdminView adminview1 = new AdminView();
            adminview.Fname = "Abhi";
            adminview.Lname = "";

            var Admin = AutoMapper.Mapper.Map<Admin>(adminview);
            if (string.IsNullOrEmpty(adminview.Fname))
            {
                response.Message = "Please provide type name";
                response.IsSuccess = false;
                return JsonConvert.SerializeObject(response);
            }
            if (string.IsNullOrEmpty(adminview.Lname))
            {
                response.Message = "Please provide type name";
                response.IsSuccess = false;
                return JsonConvert.SerializeObject(response);
            }
            if (string.IsNullOrEmpty(adminview.Email))
            {
                response.Message = "Please provide type name";
                response.IsSuccess = false;
                return JsonConvert.SerializeObject(response);
            }
            if (string.IsNullOrEmpty(adminview.Designation))
            {
                response.Message = "Please provide type name";
                response.IsSuccess = false;
                return JsonConvert.SerializeObject(response);

            }
            if (string.IsNullOrEmpty(adminview.Gender))
            {
                response.Message = "Please provide type name";
                response.IsSuccess = false;
                return JsonConvert.SerializeObject(response);
            }
            if (context.Admins.Any(c => c.Email == adminview.Email))
            {
                response.Message = "Please Provide Another Email id";
                response.IsSuccess = false;
                return JsonConvert.SerializeObject(response);
            }
            context.Admins.Add(Admin);
            context.SaveChanges();
            response.Message = "Record Save sucessfully!";
            response.IsSuccess = true;

            return JsonConvert.SerializeObject(response);
        }     

        [HttpPost]
        public string Edit(string Email)
        {
            var admin = context.Admins.Find(Email);
            return JsonConvert.SerializeObject(AutoMapper.Mapper.Map<AdminView>(admin));
        }
        [HttpPost]
        public string UpdateLeave(Int64 AlocatedLeaves)
        {
            var adminlist = context.Admins.ToList();
          
            foreach (var admin in adminlist)
            {

                var ab = AlocatedLeaves - admin.AlocatedLeaves;
                admin.CurrentBalance = admin.CurrentBalance+ ab;
                admin.AlocatedLeaves = AlocatedLeaves;
               

                context.Entry(admin).State = System.Data.Entity.EntityState.Modified;
                context.SaveChanges();
                response.IsSuccess = true;
                response.Message = "Record updated succesfully";
            }
               return JsonConvert.SerializeObject(response);
        }
        [HttpPost]
        public string Update(AdminView adminview)
        {
            if (string.IsNullOrEmpty(adminview.Fname))
            {
                response.Message = "Please provide TYpe name";
                response.IsSuccess = false;
                return JsonConvert.SerializeObject(response);
            }
            if (string.IsNullOrEmpty(adminview.Lname))
            {
                response.Message = "Please provide TYpe name";
                response.IsSuccess = false;
                return JsonConvert.SerializeObject(response);
            }
            if (string.IsNullOrEmpty(adminview.Email))
            {
                response.Message = "Please provide TYpe name";
                response.IsSuccess = false;
                return JsonConvert.SerializeObject(response);
            }
            if (string.IsNullOrEmpty(adminview.Designation))
            {
                response.Message = "Please provide TYpe name";
                response.IsSuccess = false;
                return JsonConvert.SerializeObject(response);
            }
            var Admin = AutoMapper.Mapper.Map<Admin>(adminview);
            context.Entry(Admin).State = System.Data.Entity.EntityState.Modified;
            context.SaveChanges();
            response.IsSuccess = true;
            response.Message = "Record updated succesfully";

            return JsonConvert.SerializeObject(response);
        }
        [HttpPost]
        public string date(Int64 adminview,string Email)
        {

            var admins = context.Admins.Find(Email);
            if (admins != null)
            {
                admins.CurrentBalance= adminview;
                context.Entry(admins).State = System.Data.Entity.EntityState.Modified;
                context.SaveChanges();
                response.IsSuccess = true;
                response.Message = "Record updated succesfully";
            }
            return JsonConvert.SerializeObject(response);
        }
        [HttpGet]
        public string Delete(Int64 UserId)
        {
            var admin = context.Admins.Find(UserId);
            if (admin != null)
            {
                context.Admins.Remove(admin);
                context.SaveChanges();
                response.IsSuccess = true;
                response.Message = "Record deleted successfully!";
            }

            return JsonConvert.SerializeObject(response);
        }
    }
}

File: LeaveManegamentProject/LeaveManegamentProject/AppScripts/Admin/AdminController.js
app.controller("AdminController", ["$scope", "$http","$rootScope", "$stateParams", "$location", "leaveservice",  "$rootScope",
    function ($scope, $http, $rootScope, $stateParams, $location, leaveservice) {


        $scope.Admins = [];

        $scope.GetAllData = function () {
            $http.get('/Admin').success(function (response) {
                console.log(response);
                $scope.Admins = response;

            });
        };
        $scope.GetAllData();
        $scope.Delete = function (UserId) {

            $http.get("/Admin/Delete/" + UserId).success(function (response) {

                if (response.IsSuccess) {
                    leaveservice.RecordStatusMessage("success", response.Message);
                    $scope.GetAllData();
                }
                else {
                    leaveservice.RecordStatusMessage("danger", response.Message);
                }
            });
        };
        if ($stateParams.ProjectName != undefined) {

            $scope.ProjectName = $stateParams.ProjectName;
            console.log($scope.ProjectName);
            return ($scope.ProjectName);
        }
        $rootScope.$on("tablerefresh", function (){
            $scope.GetAllData();

        });
        
    }

]);

app.controller("AdminAddController", ["$scope", "$http", "$rootScope", "$stateParams", "$location", "leaveservice",
    function ($scope, $http, $rootScope, $stateParams, $location, leaveservice) {

        $scope.Admin = {};
        $scope.IsEditMode = false;
        $scope.IsVisible = false;
        $scope.Visible = false;
        console.log($stateParams);
    

        if ($stateParams.Email != undefined) {
            $scope.IsEditMode = true;
            $http.post("/Admin/edit", { Email: $stateParams.Email }).success(function (response) {
                console.log(response);
                console.log($stateParams);       
              
               
                $scope.Admin = response;
            });
           
            
        }
        //if ($stateParams.email != undefined) {

        //    $scope.IsEditMode = true;
        //    $http.post("/Admin/Login", { email: $stateParams.email }).success(function (response) {
        //        console.log(response);
        //        console.log($stateParams);


        //        if (response.IsSuccess) {
        //            leaveservice.RecordStatusMessage("success", response.Message);
        //            $scope.Admin = response.Data;
        //            console.log($scope.Admin.role);

        //            $scope.IsVisible = "Admin";
        //            console.log($scope.IsVisible);
        //            if ($scope.Admin.role == $scope.IsVisible) {
        //                $scope.IsVisible = $scope.IsVisible = true;
        //            }
        //            else {
        //                $scope.IsVisible = $scope.IsVisible = false;
        //            }
        //            return $scope.Admin;
        //        } else {
        //            $location.path("/Admin/home");
        //            leaveservice.RecordStatusMessage("error", response.Message);
        //        }
        //    });


        //}

     
        $scope.Login = function (Admin) {
          
            $stateParams.Email = Admin.Email;
            //  $stateParams.password = Admin.password;
            password = Admin.password;
            
            if ($stateParams.Email != undefined) {
                $http.post("/Admin/Login", { Email: $stateParams.Email, password }).success(function (response) {
                    console.log(response);
                    console.log($stateParams);
                   
                    if (response.IsSuccess) {
                        $rootScope.$emit("CallParentMethod", {});
                        leaveservice.RecordStatusMessage("success", response.Message);
                        $scope.Admin = response.Data;
                       
                        $location.path("/leave/" + Admin.Email);
                        //$scope.IsVisible = "Admin";
                        //$scope.IsVisible = "Team Lead";
                        //console.log($scope.IsVisible);
                       // if ( $scope.Admin.Role == $scope.IsVisible) {
                       //     $scope.IsVisible = true;
                       // }
                       // else {
                       //     $scope.IsVisible = $scope.IsVisible = false;
                       // }
                       //// return $scope.IsVisible;
                    } else {
                        $location.path("/Admin/home");
                        leaveservice.RecordStatusMessage("error", response.Message);
                    }
                });
            }
        }
        //$rootScope.$on("MyFunction", function (CurrentBalance) {
        //    $scope.CurrentBalance;
        //    alert('Ctrl1 MyFunction')
        //    console.log($scope.CurrentBalance);
        //    if ($scope.IsEditMode = true) {
        //        $http.post("/Admin/update", { adminview: $scope.Admin }).success(function (response) {
        //            console.log(response);
        //            if (response.IsSuccess) {
        //                $scope.Admin = {};
        //                leaveservice.RecordStatusMessage("success", response.Message);
        //                $location.path("/admin");

        //            } else {
        //                leaveservice.RecordStatusMessage("error", response.Message);
        //            }


        //        });
        //    }
        //});
        $scope.Save = function () {
            if ($scope.IsEditMode) {
               // $scope.Admin.CurrentBalance = $scope.Admin.AlocatedLeaves;
                $http.post("/Admin/update", { adminview: $scope.Admin }).success(function (response) {
                    console.log(response);
                    if (response.IsSuccess) {
                        $scope.Admin = {};
                        leaveservice.RecordStatusMessage("success", response.Message);
                        $location.path("/admin");

                    } else {
                        leaveservice.RecordStatusMessage("error", response.Message);
                    }

                });
            }
            else {
                $scope.Admin.CurrentBalance = $scope.Admin.AlocatedLeaves;
                $http.post("/Admin/Save", { adminview: $scope.Admin }).success(function (response) {
                    console.log(response);
                    if (response.IsSuccess) {
                        $scope.Admin = {};
                        leaveservice.RecordStatusMessage("success", response.Message);
                        $location.path("/admin");

                    } else {
                        leaveservice.RecordStatusMessage("error", response.Message);

                    }
                    return $scope.Admin;
                });

            }

        };
        $scope.UpdateLeave = function (Admin) {
            AlocatedLeaves = Admin;
            $http.post("/Admin/UpdateLeave", { AlocatedLeaves}).success(function (response) {
                if (response.IsSuccess) {
                    $rootScope.$emit("tablerefresh", {  });
                    leaveservice.RecordStatusMessage("success", response.Message);
                } else {
                    leaveservice.RecordStatusMessage("error", response.Message);
                }
                

            });
        }
    }
    
]);